integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/Pages/MetalakePage.java

```diff
@@ -0,0 +1,260 @@
+    WebElement deleteMetalakeBtn = driver.findElement(By.xpath(xpath));
+    deleteMetalakeBtn.click();
+  }
+
+  public void clickViewMetalakeBtn(String name) {
+    String xpath = "//button[@data-refer='view-metalake-" + name + "']";
+    WebElement viewMetalakeBtn = driver.findElement(By.xpath(xpath));
+    viewMetalakeBtn.click();
+  }
+
+  public void clickEditMetalakeBtn(String name) {
+    String xpath = "//button[@data-refer='edit-metalake-" + name + "']";
+    WebElement editMetalakeBtn = driver.findElement(By.xpath(xpath));
+    editMetalakeBtn.click();
+  }
+
+  public void setMetalakeProps(int index, String key, String value) {
+    String keyPath = "//div[@data-refer='add-props-key-" + index + "']//input[@name='key']";
+    WebElement keyInput = driver.findElement(By.xpath(keyPath));
+    keyInput.sendKeys(key);
+
+    String valuePath = "//div[@data-refer='add-props-value-" + index + "']//input[@name='value']";
+    WebElement valueInput = driver.findElement(By.xpath(valuePath));
+    valueInput.sendKeys(value);
+  }
+
+  public void waitElementDisplayed(String name) {
+    int retry = 0;
+    int sleepTimeMillis = 1_000;
+
+    while (retry++ < 3) {
+      try {
+        WebElement ele = findElementByLink(name);
+        Wait<WebDriver> wait =
+            new FluentWait<>(driver)
+                .withTimeout(Duration.ofSeconds(2))
+                .pollingEvery(Duration.ofMillis(300))
+                .ignoring(ElementNotInteractableException.class);
+        wait.until(
+            d -> {
+              ele.isDisplayed();
+              return true;
+            });
+        Thread.sleep(sleepTimeMillis);
+      } catch (Exception e) {
+        LOG.error(e.getMessage(), e);
+      }
+    }
+  }
+
+  public boolean verifyCreateMetalake(String name) {
+    try {
+      String rowPath = "//div[@data-id='" + name + "']";
+      WebElement createdMetalakeRow = driver.findElement(By.xpath(rowPath));
+      boolean isRow = createdMetalakeRow.isDisplayed();
+
+      String linkPath = "//div[@data-field='name']//a[@href='/ui/metalakes?metalake=" + name + "']";
+      WebElement createdMetalakeLink = driver.findElement(By.xpath(linkPath));
+      boolean isLink = createdMetalakeLink.isDisplayed();
+      boolean isText = Objects.equals(createdMetalakeLink.getText(), name);
+
+      return isRow && isLink && isText;
+    } catch (Exception e) {
+      return false;
+    }
+  }
+
+  public boolean verifyEditedMetalake(String name) {
+    try {
+      String xpath = "//div[@data-field='name']//a[@href='/ui/metalakes?metalake=" + name + "']";
+      WebElement editedMetalakeLink = driver.findElement(By.xpath(xpath));
+
+      return Objects.equals(editedMetalakeLink.getText(), name);
+    } catch (Exception e) {
+      return false;
+    }
+  }
+
+  public boolean verifyShowMetalakeDetails() {
+    try {
+      metalakeDetailsDrawer.isDisplayed();
+      String drawerVisible = metalakeDetailsDrawer.getCssValue("visibility");
```

Comment:
It's too sketchy to just check if it's visible or not.
Don't we need check value on each element?